contracts/core/EntryPoint.sol

```diff
@@ -110,7 +118,9 @@ contract EntryPoint is IEntryPoint, StakeManager {
             IAggregator aggregator = opa.aggregator;
             uint256 opslen = ops.length;
             for (uint256 i = 0; i < opslen; i++) {
-                _validatePrepayment(opIndex, ops[i], opInfos[opIndex], address(aggregator));
+                UserOpInfo memory opInfo = opInfos[opIndex];
+                (uint256 deadline, uint256 paymasterDeadline,) = _validatePrepayment(opIndex, ops[i], opInfo, address(aggregator));
+                _validateDeadline(i, deadline, opInfo.mUserOp.paymaster, paymasterDeadline);
                 opIndex++;
```

Comment:
We report the index (in failop), so I didn't want to use type(uint).max ...
Can we assume that 1 million UserOps in a bundle is beyond the possible capacity of ethereum evm ? (raised from 100,000)

Reply:
See my other comment about this number.
1. On a normal chain a million is well beyond reach.  But what about a zkEVM where blocks are not submitted, and the prover just proves the validity of the state transition?  I'm no ZK expert, but some of these chains prove huge state changes every few hours, spanning across a many blocks that happened during that time.  They could theoretically support very large blocks.
2. Why is this number even needed?  See my suggestion regarding aggregator==address(EntryPoint).

Reply:
didn't find this aggregator==address(entrypoint).
the index appears in FailedOp, and dumped in logs.
Do you think `handleOps(ops[])` can be executed on a zkEVM with an array of over million entries?

Reply:
I don't know why you can't see my comments.  It appears like you hid them (unanswered) and I can only unhide them on my end.  I'll send you separately.

I don't know whether a zkEVM prover will handle an array with over a million entries or not, and whether it can't do it now but will be able to do it in a couple of years.  zkEVM is making rapid progress.  I'm pretty sure it won't be able to do 2**256-1 but I'm not so sure about 1M.  They already prove a few hours of state transitions in a single proof, so why not a single large block?

But regardless, I'm not sure why this number is needed.  I wrote about it in one of the hidden comments.

Reply:
We are not talking about proving hours of data - we specifically talk about a single handleOps call, and a batch of UserOps in it - and this batched (single) transaction we have index for. it can't span between different transactions.